1 Lesson/Homework/Garden.rb

```diff
@@ -0,0 +1,145 @@
+class Raspberry
+  @@states = [:Отсутствует, :Цветение, :Зеленая, :Красная]
+
+  def initialize(index)
+    @index = index
+    @state = @@states[0]
+  end
+
+  def grow!
+    if @state != :Красная
+      @state = @@states[(@@states.find_index(@state) + 1) % @@states.size]
+    end
+  end
+
+  def ripe?
+    @state == :Красная
+  end
+
+  def grab!
+    @state = @@states[0]
+  end
+end
+
+class RaspberryBush
+  attr_reader :berries_quantity
+
+  def initialize(berries_quantity)
+    @berries_quantity = berries_quantity
+    @raspberries = berries_quantity.times.map { |i| Raspberry.new(i) }
+  end
+
+  def grow_all!
+    @raspberries.each { |berry| berry.grow! }
+  end
+
+  def ripe_all?
+    @raspberries.each { |berry| return false unless berry.ripe? }
+    true
+  end
+
+  def give_away_all!
+    @raspberries.each { |berry| berry.grab! }
+  end
+end
+
+class Human
+  attr_reader :name
+  attr_reader :money
+  attr_reader :statistic_of_harvested_berries
+
+  def initialize(name, plant)
+    @name = name
+    @plant = plant
+    @amount_of_harvested_berries = 0
+    @statistic_of_harvested_berries = 0
+    @money = 0
+  end
+
+  def work!
+    print "Дачник по имени #{@name} начал работу\n"
+    @plant.grow_all!
+  end
+
+  def harvest!
+    if @plant.ripe_all?
+      @amount_of_harvested_berries += @plant.berries_quantity
+      @statistic_of_harvested_berries += @plant.berries_quantity
```

Comment:
Looks not optimized :)
Are you sure that you want to collect statistics here?

Reply:
I changed trigger for summing up @statistic_of_harvested_berries, now it sums up from @amount_of_harvested_berries when worker decides to sell them(Human.sell_berries!) and @statistic_of_harvested_berries is recieved through function
```
def statistic_of_harvested_berries
  @statistic_of_harvested_berries + @amount_of_harvested_berries
end
```
```suggestion
      @amount_of_harvested_berries += @plant.berries_quantity
```